lib/sync.py

```diff
@@ -126,9 +126,7 @@ def extend_transactions(self, transactions, payments, syncpair):
             transfer_to = next((sp for sp in self.syncpairs
                                 if sp["iban"] == p["iban"]), None)
             transaction = ynab.get(import_id)
-            if transaction:
-                transaction["payment"] = p
-            else:
+            if not transaction or transaction["cleared"] == "uncleared":
```

Comment:
As I read it, this change will try to create a new transaction at YNAB's end for an uncleared transaction that already exists.

Wouldn't this just bounce off YNAB's duplicate detection?

Reply:
You read it right.

As I have experienced it, this effectively works as either a PUT (duplicate) or a PATCH, depending.

I double checked the API following what you said and I realise that there is indeed an actual PATCH that would be the better semantic approach here. I'll review my patch later.

Reply:
(but in actuality, this patch works as intended and will allow ynab to register the "matched" part of the bunq transaction.)

Reply:
Fixed behaviour with new patch. I now leverage the "dirty" flag instead.